lib/ret/app_config.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Ret.AppConfig do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Ret.{AppConfig, Repo}
+
+  @schema_prefix "ret0"
+  @primary_key {:app_config_id, :id, autogenerate: true}
+
+  schema "app_configs" do
+    field(:key, :string)
+    field(:value, :map)
+    belongs_to(:owned_file, Ret.OwnedFile, references: :owned_file_id)
+    timestamps()
+  end
+
+  def changeset(%AppConfig{} = app_config, attrs) do
+    # We wrap the config value in an outer %{value: ...} map because we want to be able to accept primitive
+    # value types, but store them as json.
+    attrs = attrs |> Map.put("value", %{value: attrs["value"] |> Poison.decode!()})
+
+    app_config
+    |> cast(attrs, [:key, :value])
+    |> unique_constraint(:key)
+  end
+
+  def get_config() do
```

Comment:
might be premature optimization, but putting this behind a cachex key seems wise